jina/peapods/gateway.py

```diff
@@ -144,6 +144,44 @@ def prefetch_req(num_req, fetch_to):
                     prefetch_task = [j for j in onrecv_task]
 
 
+class AsyncGatewayPea:
+    def __init__(self, args):
+        if not args.proxy and os.name != 'nt':
+            os.unsetenv('http_proxy')
```

Comment:
This feels like users may get very mad at us?

Reply:
Why is this needed?

Reply:
Agree. Unsetting `http_proxy` and `https_proxy` sounds really bad.

Reply:
it is needed since GNES @deepankarm @nan-wang @JoanFM please 
1. read the old code
2. when you running grpc with http and https proxy you will find grpc won't work anymore.
3. always keep in mind os.environ is set on process base not globally, there is nothing people get mad at us, because nothing change in global 
4. anytime, use warning.warn

Reply:
please always understand grpc is on HTTP2 layer and that's why http proxy setting affects it. if you don't know about this, read grpc doc.

if you really want to solve this elegantly, i suggest looking for issue help/posting issue on grpc github.

also please fix the misconception that os.environ changes anything globally.

until then, keep this lines. Many companies have internal proxy, without this line, Jina won't work for them.

Reply:
Are we sure this init lives in another process?

Reply:
Shouldn't it be moved into start?